pennylane/transforms/decompose.py

```diff
@@ -838,10 +839,32 @@ def _operator_decomposition_gen(  # pylint: disable=too-many-arguments
     if max_expansion is not None and max_expansion <= current_depth:
         max_depth_reached = True
 
-    if acceptance_function(op) or max_depth_reached:
+    # Handle MCMs
+    if isinstance(op, measurements.MidMeasureMP):
         yield op
+
+    # Handle classically controlled operators
+    elif isinstance(op, Conditional):
+        if acceptance_function(op.base) or max_depth_reached:
+            yield op
+        else:
+            decomp = [
+                Conditional(op.meas_val, base_op)
+                for base_op in _operator_decomposition_gen(
+                    op.base,
+                    acceptance_function,
+                    max_expansion=max_expansion,
+                    current_depth=current_depth,
+                    graph_solution=graph_solution,
+                )
+            ]
```

Comment:
This should just work out of the box with the non-graph plxpr and tape transforms, right? If yes, this needs testing for both the plxpr transform and tape tansform.